repos/system_upgrade/common/libraries/mdraid.py

```diff
@@ -0,0 +1,41 @@
+from leapp.exceptions import StopActorExecution
+from leapp.libraries.stdlib import api, CalledProcessError, run
+
+
+def is_mdraid_dev(dev):
+    """
+    Check if a given device is an md (Multiple Device) device
+
+    :raises: StopActorExecution in case of error
+    """
+    try:
+        result = run(['mdadm', '--query', dev])
+    except CalledProcessError as err:
+        api.current_logger().warning(
+                'Could not check if device is an md device: {}'.format(err)
+        )
+        raise StopActorExecution()
```

Comment:
@matejmatuska this is wrong. this function is in public library, which means it can be called by whatever actor. And right now, it's expected the function return True when it is raid device, or false when is not. In systems without the raid, it's possible typical that `mdadm` is not installed, in such a case the function should return False. Stopping an actor execution should be done inside the actor - not inside the public library function in this case (unless we speak about hard error, which this is not).

Reply:
probably best solution would be to check whether mdadm is installed. in case it is installed and the call crashes, than probably would make sense to actually expose the calledprocesserror and keep the handlin on an actor.

Reply:
Should be resolved now